### ISSUE TITLE
Incorrect config does not cause pyright to give exit code 3

### ISSUE THREAD

**Issue by hth313** [microsoft/pyright#4722]
**Describe the bug**
The documentation states that I should get exit code 3 if the config file cannot be read or parsed. I see error code 0.
I tried with very broken json, but it seems hard to provoke. The reason I want to do this is for testing an invocation wrapper of pyright and I want a situation where it fails to run.

**To Reproduce**
Add a broken config file, I use:
```
{
  "include": 10,

  "exclude": 2,

  "ignore": [
    "src/oldstuff"
  ],

  "defineConstant": {
    "DEBUG": true
  },
}
```
Stand in the directory and do `pyright .`, among other things I get:
```
Loading configuration file at ....fail_python/pyrightconfig.json
Config "exclude" entry must contain an array.
```
Now print the exit code:
```
$ echo $?
0
```

**Expected behavior**
I would expect to see exit code 3.

**Screenshots or Code**
No source code needed.

**VS Code extension or command-line**
pyright 1.1.294

**Comment by erictraut** [comment]
This is working as designed. The example you provided is valid JSON. If you want to generate an exit code of 3, you should use invalid JSON syntax.

**Comment by hth313** [comment]
OK, so I have used  `{ {` and it silently accepts it.

**Comment by erictraut** [comment]
A while back, we switched from the normal JSON parser to a more permissive JSONC parser to support comments and hanging commas within the config file. By default, the JSONC parser recovers the parse when it encounters a syntax error rather than throwing an exception like the normal JSON parser. I've updated the code to tell the JSONC parser to be stricter so we can detect and report syntax errors in the config file.

This will be included in the next release.
